tests/phpunit/tests/block-supports/duotone.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * Test the block WP_Duotone class.
+ *
+ * @package WordPress
+ */
+
+class WP_Duotone_Test extends WP_UnitTestCase {
+	/**
+	 * Cleans up CSS added to block-supports from duotone styles. We neeed to do this
+	 * in order to avoid impacting other tests.
+	 */
+	public static function wpTearDownAfterClass() {
+		WP_Style_Engine_CSS_Rules_Store::remove_all_stores();
+	}
+
+	/**
+	 * Tests whether the duotone preset class is added to the block.
+	 *
+	 * @ticket 58555
+	 *
+	 * @covers ::render_duotone_support
+	 */
+	public function test_render_duotone_support_preset() {
+		$block         = array(
+			'blockName' => 'core/image',
+			'attrs'     => array( 'style' => array( 'color' => array( 'duotone' => 'var:preset|duotone|blue-orange' ) ) ),
+		);
+		$block_content = '<figure class="wp-block-image size-full"><img src="/my-image.jpg" /></figure>';
+		$expected      = '<figure class="wp-block-image size-full wp-duotone-blue-orange"><img src="/my-image.jpg" /></figure>';
+		$this->assertSame( $expected, WP_Duotone::render_duotone_support( $block_content, $block ) );
+	}
+
+	/**
+	 * Tests whether the duotone unset class is added to the block.
+	 *
+	 * @ticket 58555
+	 *
+	 * @covers ::render_duotone_support
+	 */
+	public function test_render_duotone_support_css() {
+		$block         = array(
+			'blockName' => 'core/image',
+			'attrs'     => array( 'style' => array( 'color' => array( 'duotone' => 'unset' ) ) ),
+		);
+		$block_content = '<figure class="wp-block-image size-full"><img src="/my-image.jpg" /></figure>';
+		$expected      = '/<figure class="wp-block-image size-full wp-duotone-unset-\d+"><img src="\\/my-image.jpg" \\/><\\/figure>/';
+		$this->assertMatchesRegularExpression( $expected, WP_Duotone::render_duotone_support( $block_content, $block ) );
+	}
+
+	/**
+	 * Tests whether the duotone custom class is added to the block.
+	 *
+	 * @covers ::render_duotone_support
+	 */
+	public function test_render_duotone_support_custom() {
+		$block         = array(
+			'blockName' => 'core/image',
+			'attrs'     => array( 'style' => array( 'color' => array( 'duotone' => array( '#FFFFFF', '#000000' ) ) ) ),
+		);
+		$block_content = '<figure class="wp-block-image size-full"><img src="/my-image.jpg" /></figure>';
+		$expected      = '/<figure class="wp-block-image size-full wp-duotone-ffffff-000000-\d+"><img src="\\/my-image.jpg" \\/><\\/figure>/';
+		$this->assertMatchesRegularExpression( $expected, WP_Duotone::render_duotone_support( $block_content, $block ) );
+	}
+
+	public function data_get_slug_from_attribute() {
```

Comment:
Data provide should go below the test and have the docblock

```
/**
 * Data provider.
 *
 * @return array[].
 */
```

:1234: Applies below too